tiledb/sm/array_schema/CMakeLists.txt

```diff
@@ -54,7 +54,7 @@ conclude(object_library)
 #
 commence(object_library domain)
     this_target_sources(domain.cc)
-    this_target_object_libraries(datum dimension math)
+    this_target_object_libraries(datum dimension math ndrectangle current_domain)
```

Comment:
Just another somehow related comment on why this problem came up. It is my understanding that `expand_to_tiles` is actually operating on a `Range` and should be a method of that class instead and not of `Domain`. I realized that when I wanted to suggested that we pass `NDRange& query_ndrange` as `const` and realized that the method's side effects are in fact in that class so we can't. This is quite old code and has been structured that way since forever so I didn't want to ask for such a large refactoring as part of this PR that is just trying to introduce a bugfix and it's also quite urgent. We could open a ticket as a follow up to restructure stuff internally so that we try to move it to `Range` from the temporarily solution of `CurrentDomain` and see if that avoids circular dependencies. In theory such circular dependencies are indicators of wrong design, and IMO that's the case here as well.

Reply:
+1

The commit https://github.com/TileDB-Inc/TileDB/pull/5390/commits/e631d93bfd6b127e71f4e0fb981986d3f3b3acee:

* Did not have circular dependencies
* Fixed the time-sensitive customer-impacting bug